controller/metrics.go

```diff
@@ -117,36 +150,46 @@ func convertTime(vm mo.VirtualMachine) float64 {
 	return float64(vm.Summary.Runtime.BootTime.Unix())
 }
 
-func powerState(s types.HostSystemPowerState) float64 {
-	if s == "poweredOn" {
+func powerState(s interface{}) float64 {
```

Comment:
As more specific typing is generally preferred where possible, I'm assuming something warrants the switch to an interface?

Reply:
I wanted to use `powerState` to return values from two different types (`HostSystemPowerState` and `VMPowerState`, I believe.) and `interface` makes it easier to make it generic.  Off the top of my head, I'm not sure if there is a way to have the function accept only those two types or not, but I can look into it.

Reply:
I left the generic `interface` but added some type assertions in the function to prevent strange bugs.  Check it out and let me know what you think.